docs/assert/timeout.md

```diff
@@ -0,0 +1,36 @@
+---
+layout: default
+title: timeout
+description: Sets the length of time to wait for async operations before failing the test.
+categories:
+  - assert
+  - async
+---
+
+## `timeout( duration )`
+
+Sets the length of time to wait for async operations before failing the test.
+
+| name | description |
+|------|-------------|
+| `duration` (Number) | The length of time, in milliseconds, to wait for async operations. |
+
+### Description
+
+`assert.timeout()`sets the length of time, in milliseconds, to wait for async operations in the current test. This is equivalent to setting `config.testTimeout` on a per-test basis. The timeout length only applies when performing async operations.
+
+If a non-numeric value is passed as an argument, the function will throw an error.
+
+### Examples
+
+```js
+QUnit.test( "async test", function( assert ) {
+  assert.timeout( 1000 ); // Timeout of 1 second
+  var done = assert.async();
+  var input = $( "#test-input" ).focus();
+  setTimeout(function() {
+    assert.equal( document.activeElement, input[0], "Input was focused" );
```

Comment:
Agree with @gibson042 here. Note: Since the above comment was added, a non-DOM example was also added below, which is great.

I don't particularly mind having an example involving DOM in someway, but as unimportant as it might seem, the specific example we have above seems like an anti-pattern I'd rather not promote in the documentation.

I'd be okay with having just the below example, although in that case I'd rather have the main example showcase `assert.async()` which is easier to understand when learning QUnit. Having two examples, where one shows compat with Promise still would be nice too, though.